pandas/io/tests/parser/skiprows.py

```diff
@@ -198,3 +199,21 @@ def test_skiprows_infield_quote(self):
 
         df = self.read_csv(StringIO(data), skiprows=2)
         tm.assert_frame_equal(df, expected)
+
+    def test_skiprows_callable(self):
+        data = 'a\n1\n2\n3\n4\n5'
+
```

Comment:
Can you add tests with bad functions (raises and returns non-bool) - in particular want to make sure error propagates from c-engine.